ganalytics.cpp

```diff
@@ -473,10 +501,12 @@ GAnalytics::GAnalytics(const QString &trackingID, QObject *parent)
 }
 
 /**
- * Destructor of class GAnalytics.
+ * Destructor of class GAnalytics. Ends the session.
  */
 GAnalytics::~GAnalytics()
 {
+    if (!trackingID().isEmpty())
+        endSession();
```

Comment:
No, I think you were right before your edit: isSending is only sent to false after all the messages were sent or if there was an error send ing the messages. In both cases its ok to quit, I think.